raster/src/main/scala/geotrellis/raster/geotiff/GeoTiffReprojectRasterSource.scala

```diff
@@ -112,22 +112,31 @@ class GeoTiffReprojectRasterSource(
       queryPixelBounds <- bounds
       targetPixelBounds <- queryPixelBounds.intersection(this.dimensions)
     } yield {
+      val targetExtent = gridExtent.extentFor(targetPixelBounds)
       val targetRasterExtent = RasterExtent(
-        extent = gridExtent.extentFor(targetPixelBounds, clamp = true),
+        extent = targetExtent,
+        cols = targetPixelBounds.width.toInt,
+        rows = targetPixelBounds.height.toInt
+      )
+
+      // buffer the targetExtent to read a buffered area from the source tiff
+      // so the resample would behave properly on borders
+      val bufferedTargetRasterExtent = RasterExtent(
```

Comment:
This is RasterExtent for no reason? You're just buffering the extent and then back-projecting that. Just a little confusing.